massa-ledger/src/ledger.rs

```diff
@@ -184,4 +195,231 @@ impl FinalLedger {
             .get(addr)
             .map_or(false, |v| v.datastore.contains_key(key))
     }
+
+    /// Get a part of the ledger
+    /// Used for bootstrap
+    /// Parameters:
+    /// * cursor: Where we stopped in the ledger
+    ///
+    /// Returns:
+    /// A subset of the ledger starting at `cursor` and of size `LEDGER_PART_SIZE_MESSAGE_BYTES` bytes.
+    pub fn get_ledger_part(
+        &self,
+        cursor: Option<LedgerCursor>,
+    ) -> Result<(Vec<u8>, LedgerCursor), ModelsError> {
+        let mut next_cursor = cursor.unwrap_or(LedgerCursor(
+            *self
+                .sorted_ledger
+                .first_key_value()
+                .ok_or_else(|| ModelsError::BufferError("Ledger empty".into()))?
```

Comment:
the ledger being empty should not crash the function, just return a finish

Reply:
I would add a warning too :slightly_smiling_face:

Reply:
I have turn the return type to an option of cursor and if there is no ledger it will return an empty data and none as a cursor.